folly/Traits.h

```diff
@@ -312,6 +312,36 @@ struct is_negative_impl<T, false> {
   constexpr static bool check(T x) { return false; }
 };
 
+// Early versions of MSVC choke on the template
+// constraints.
+#if _MSC_FULL_VER <= 200022816 // 2015 RC or below
```

Comment:
Unfortunately it does still need it. I believe that is the version number for the RTM release, so the comment must not have gotten updated.